### PR TITLE
Node registry REST API refactor

### PR DESCRIPTION
**Pull request by ltseeley**
Cleans up and refactors the node registry REST API code to be up to the standards of the other REST API code that has undergone stabilization.

### CODE REVIEW

File: libsplinter/src/node_registry/mod.rs [hunk 1]
```diff
@@ -254,6 +254,12 @@ pub trait RwNodeRegistry: NodeRegistryWriter + NodeRegistryReader {
     ///  }
     ///```
     fn clone_box(&self) -> Box<dyn RwNodeRegistry>;
+
```
Comment (agunde406) [inline libsplinter/src/node_registry/mod.rs:261]: Commit message says "Also fixes a typo in the `clone_box` doc string for `RwNodeRegistry`." but it doesnt look like that is actually included in this commit.

File: splinterd/api/static/openapi.yml [hunk 2]
```diff
@@ -401,9 +401,13 @@ paths:
 
   /admin/nodes:
     post:
+      summary: Add a node to the registry
+      description: |
+        This endpoint can be used to a new node to the node registry. The node
```
Comment (agunde406) [inline splinterd/api/static/openapi.yml:406]: Typo: "a new node -> "add a new node"

File: libsplinter/src/node_registry/mod.rs (hunk 1 above)
Comment (ltseeley) [reply libsplinter/src/node_registry/mod.rs:261]: Removed from the commit message

File: splinterd/api/static/openapi.yml (hunk 2 above)
Comment (ltseeley) [reply splinterd/api/static/openapi.yml]: Fixed

**Pull request merged**